### PR TITLE
Added the module Deploying a Camel Route in AWS Lambda to the examples

### PR DESCRIPTION
**Pull request by ravishankarhassain**
Note that the `main` branch points at the latest stable Camel Quarkus release.
Pull requests should be generally send against the `camel-quarkus-main` branch pointing at the current Camel Quarkus SNAPSHOT.

### CODE REVIEW

**Comment by davsclaus** [comment]
The java source code files need ASF license headers, and so does the pom.xml file etc. See the other examples

**Comment by ravishankarhassain** [comment]
@davsclaus : I have updated the source code to have ASF license headers

**Comment by davsclaus** [comment]
Maybe the example should just be called `aws-lambda` as that is the main focus, not really so much direct log cdi components.

**Comment by ravishankarhassain** [comment]
@davsclaus : I have renamed the example as aws-lambda

**Review by davsclaus — APPROVED** [review]

File: aws-lambda/src/test/java/io/quarkus/camel/AWSLambdaHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import javax.inject.Inject;
+
+@QuarkusTest
+public class AWSLambdaHandlerTest {
+
+    @Inject
```
Comment (aldettinger) [inline aws-lambda/src/test/java/io/quarkus/camel/AWSLambdaHandlerTest.java:32]: It's really great to have a test for maintenance purpose :+1: And in such situation, it should be possible to write less code with [@InjectMock](https://camel.apache.org/blog/2020/10/mocking-beans-with-camel-quarkus/)

**Review by aldettinger — CHANGES_REQUESTED** [review]
Great contribution. A few polish and it will be more aligned with other examples :)

File: aws-lambda/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+*
```
[the remark this answers is not in the document]
Comment (ravishankarhassain) [reply aws-lambda/.dockerignore]: Done

File: aws-lambda/.gitignore [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#Maven
```
[the remark this answers is not in the document]
Comment (ravishankarhassain) [reply aws-lambda/.gitignore]: Done

File: aws-lambda/mvnw [hunk 4]
```diff
@@ -0,0 +1,310 @@
+#!/bin/sh
```
[the remark this answers is not in the document]
Comment (ravishankarhassain) [reply aws-lambda/mvnw]: Done

File: aws-lambda/pom.xml [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>io.quarkus.camel</groupId>
+  <artifactId>aws-lambda</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+  <properties>
```
[the remark this answers is not in the document]
Comment (ravishankarhassain) [reply aws-lambda/pom.xml:27]: Updated the pom based on the example.

File: aws-lambda/pom.xml [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+      <groupId>org.apache.camel.quarkus</groupId>
+      <artifactId>camel-quarkus-direct</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.camel.quarkus</groupId>
+      <artifactId>camel-quarkus-log</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.camel.quarkus</groupId>
+      <artifactId>camel-quarkus-bean</artifactId>
```
[the remark this answers is not in the document]
Comment (ravishankarhassain) [reply aws-lambda/pom.xml:65]: Done

File: aws-lambda/src/main/docker/Dockerfile.legacy-jar [hunk 7]
```diff
@@ -0,0 +1,51 @@
+####
```
[the remark this answers is not in the document]
Comment (ravishankarhassain) [reply aws-lambda/src/main/docker/Dockerfile.legacy-jar]: Done

File: aws-lambda/src/main/java/io/quarkus/camel/CamelRoute.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+public class CamelRoute extends RouteBuilder {
+
+   @Inject
+   GreetService greetService;
+
+    @Override
+    public void configure() throws Exception {
+        from("direct:input").routeId("Test")
+           .log("Inside Camel Route Received Payload ==> ${body}")
+           /*
```
[the remark this answers is not in the document]
Comment (ravishankarhassain) [reply aws-lambda/src/main/java/io/quarkus/camel/CamelRoute.java]: Done

File: aws-lambda/src/main/resources/application.properties [hunk 9]
```diff
@@ -0,0 +1,16 @@
+## ---------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (ravishankarhassain) [reply aws-lambda/src/main/resources/application.properties]: Done

**Comment by ravishankarhassain** [comment]
@aldettinger: I have updated the based on your comments

File: aws-lambda/pom.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (aldettinger) [reply aws-lambda/pom.xml:27]: Logically, it should be possible to remove the quarkus.platform.* properties. I even think [copy pasting properties](https://github.com/apache/camel-quarkus-examples/blob/main/timer-log-cdi/pom.xml#L30..#L47) from anoher example would be quicker.

File: aws-lambda/pom.xml [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+      <artifactId>quarkus-test-amazon-lambda</artifactId>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>${quarkus.platform.groupid}</groupId>
+        <artifactId>quarkus-maven-plugin</artifactId>
+        <version>${quarkus.platform.version}</version>
```
Comment (aldettinger) [inline aws-lambda/pom.xml:96]: And here quarkus-maven-plugin would be `${quarkus.version}`, so platform is not needed.

File: aws-lambda/pom.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (ravishankarhassain) [reply aws-lambda/pom.xml]: Removed

File: aws-lambda/pom.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (ravishankarhassain) [reply aws-lambda/pom.xml:27]: updated the pom.xml

**Comment by ravishankarhassain** [comment]
@aldettinger: I have updated the pom.xml based on your comments

File: aws-lambda/pom.xml (hunk 10 above)
Comment (ravishankarhassain) [reply aws-lambda/pom.xml]: updated the pom.xml

**Comment by aldettinger** [comment]
@ravishankarhassain There was some remaining build polishing issues that I should be able to fix [there](https://github.com/apache/camel-quarkus-examples/pull/53). 

Well done for this interesting contribution :+1: I'm thinking maybe you would like to spread the word about this in a camel blog post. It's a matter of opening a PR against [camel-website](https://github.com/apache/camel-website), taking another blog post [as example](https://github.com/apache/camel-website/commit/118efee70bd8e9d4d9b5fa79e1e28b52ec4ad485), then a note will be added in the PR with a preview of the website in order to check the content like in this [PR](https://github.com/apache/camel-website/pull/611).

**Comment by ravishankarhassain** [comment]
@aldettinger : I have raised the PR https://github.com/apache/camel-website/pull/612 against the camel-website

**Pull request closed without merging**